### ISSUE TITLE
Make error - fatal error: SOIL2/src/SOIL2/SOIL2.h: No such file or directory

### ISSUE THREAD

**Issue by Daeraxa** [SpartanJ/eepp#12]
So I was having a look to see if I could build ecode and from what I understand the instructions are:
- Clone eepp
- `premake4 gmake`
- `make -C make/linux`

When I do this it keeps failing on the following (on the `make -C make/linux` command):
```
vertexbuffervbo.cpp
shader.cpp
textureregion.cpp
../../src/eepp/graphics/textureregion.cpp:1:10: fatal error: SOIL2/src/SOIL2/SOIL2.h: No such file or directory
    1 | #include <SOIL2/src/SOIL2/SOIL2.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[1]: *** [eepp-shared.make:715: ../../obj/linux/debug/eepp-shared/textureregion.o] Error 1
make: *** [Makefile:97: eepp-shared] Error 2
make: Leaving directory '/home/dae/sourcebuild/eepp/make/linux'
```

I also tried with `--with-mojoal` and got the same error.

Apologies if something was wrong here or if I've done something stupid...

**Comment by SpartanJ** [comment]
Hi!
You probably forgot to pull the submodules (efsw and SOIL2 are the two submodules eepp is using).
`git clone --recurse-submodules https://github.com/SpartanJ/eepp.git`
when cloning, and already cloned you can:
`git pull --recurse-submodules`

**Comment by Daeraxa** [comment]
I did indeed! I only did a standard git clone. I have to admit I didn't notice that bit in the readme, might be worth adding a link to that section nearer `How to build it` as it is quite a way away (or moving it directly above). Or add the version with `--recurse-submodules` to the instructions as at the moment:

> Clone the repository and run on the repository root directory:

Looks like the section could be standalone, I just took that as a standard `git clone` instruction.

It did manage to compile and build ecode so :)
Thanks for the help!
